core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java

```diff
@@ -1311,8 +1311,9 @@ public void prepare(MethodContext context) {
                         continue;
                     }
                     Class propertyType = i.getPropertyType();
-                    if (ctorParamIndex == null
-                            && i.getReadMethod().getReturnType() != i.getWriteMethod().getParameterTypes()[0]) {
+                    Class<?> getterReturnType = i.getReadMethod().getReturnType();
+                    Class<?> setterParameterType = i.getWriteMethod().getParameterTypes()[0];
```

Comment:
CI seems to indicate that you need to be more defensive here. 
I am assuming it used to work because the expression was short-circuited by the first check in the `if` statement

Reply:
Yeah, I'm already working on it.